### PR TITLE
add build.yml for Github Actions

### PR DESCRIPTION
**Pull request by madhur-tandon**
Add CI for building the project on various platforms

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    
+    - name: Setup Miniconda
+      uses: conda-incubator/setup-miniconda@v1.7.0
+    
+    - name: Install Dependencies
+      shell: bash
+      run: conda install xtl xproperty nlohmann_json -c conda-forge
```
Comment (martinRenou) [inline .github/workflows/build.yml:23]:
```suggestion
      run: conda install xtl xproperty nlohmann_json cmake -c conda-forge
```

Comment (madhur-tandon) [reply .github/workflows/build.yml]: Thanks!

**Comment by martinRenou** [comment]
I wonder if the cmake used is still the one Github actions provide. Can you try

```
which cmake
```

?

**Comment by madhur-tandon** [comment]
Yeah, I guess it's the one that comes in-built i.e. `/usr/local/bin/cmake`

**Comment by martinRenou** [comment]
You might want to manually specify the  `DCMAKE_INSTALL_PREFIX`

**Comment by martinRenou** [comment]
I wonder if $CONDA_PREFIX is actually defined, can you print it?

**Comment by madhur-tandon** [comment]
> You might want to manually specify the `DCMAKE_INSTALL_PREFIX`

Yes this is a good suggestion. Not specifying this leads to the same error locally as well. But it doesn't work here 😅
Next step is to check $CONDA_PREFIX which ideally should be defined. The name of the environment is `test`.

**Comment by madhur-tandon** [comment]
Yup, it isn't defined it seems, which is strange because according to https://github.com/marketplace/actions/setup-miniconda it should be.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      shell: bash
+      working-directory: ${{runner.workspace}}/build
+      # Note the current convention is to use the -S and -B options here to specify source 
+      # and build directories, but this is only available with CMake 3.13 and higher.  
+      # The CMake binaries on the Github Actions machines are (as of this writing) 3.12
+      run: cmake $GITHUB_WORKSPACE -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX -DCMAKE_BUILD_TYPE=$BUILD_TYPE
+
+    - name: Build
+      working-directory: ${{runner.workspace}}/build
+      shell: bash
```
Comment (martinRenou) [inline .github/workflows/build.yml:42]:
```suggestion
      shell: bash -l {0}
```

You should apply this change everywhere

Comment (madhur-tandon) [reply .github/workflows/build.yml:42]: Thanks, on it.

Also, the environment from `conda info` is `None` indeed.

Comment (madhur-tandon) [reply .github/workflows/build.yml]: Wow, I guess this fixed it. I wonder what it does though.

**Pull request merged**